src/components/GoogleCalendarButton.js

```diff
@@ -26,41 +26,90 @@ export default function GoogleCalendarButton() {
-          start: {
-            dateTime: startTime,
-            timeZone: 'America/Chicago',
-          },
-          end: {
-            dateTime: endTime,
-            timeZone: 'America/Chicago',
+  const clientID = '4436581685-mcdkhnh50a6di21k15do7d4a2orq7faf.apps.googleusercontent.com'
+  // const clientSecret = 'gPwoF99zNeSrs_JXcVaNATub'
+  const APIKey = 'AIzaSyCssknAiVBtWhHE3O1wjVte829dxoxQ-pM'
+
+  sections.forEach(
+    (section) => {
+      // '2015-05-28T09:00:00-07:00'
+      const dow = section.event.dow;
+      const date = dateOfTheFirstClass[dow];
+      const startTime = `${date + section.event.start.hour}:${section.event.start.minute}:00`;
+      const endTime = `${date + section.event.end.hour}:${section.event.end.minute}:00`;
+      const event = {
+        summary: `${section.subjectId} ${section.courseId}`,
+        description: section.name,
+        location: section.event.location,
+        start: {
+          dateTime: startTime,
+          timeZone: 'America/Chicago',
+        },
+        end: {
+          dateTime: endTime,
+          timeZone: 'America/Chicago',
+        },
+        recurrence: [
+          'RRULE:FREQ=WEEKLY;COUNT=10',
+        ],
+      };
+      events.push(event);
+    },
+  );
+
+  const initAuth = () => {
+    window.gapi.load('auth2', () => {
+      window.gapi.auth2.init({
+        client_id: clientID,
+      }).then(() => {
+        window.gapi.load('client', () => {
+          window.gapi.client.setApiKey(APIKey);
+          window.gapi.client.load('https://content.googleapis.com/discovery/v1/apis/calendar/v3/rest')
+            .then(() => { console.log('GAPI client loaded for API'); },
+              (err) => { console.error('Error loading GAPI client for API', err); });
+        });
+        window.gapi.auth2.getAuthInstance()
+          .signIn({ scope: 'https://www.googleapis.com/auth/calendar https://www.googleapis.com/auth/calendar.events'})
+          .then(() => {
+            console.log('Sign-in successful');
+            events.forEach(
+              (event) => {
+                const newEvent = {
+                  calendarId: 'primary',
+                  resource: event,
+                };
+                window.gapi.client.calendar.events.insert(newEvent)
+                  .then((response) => {
+                    console.log('Response', response);
+                  },
+                  (err) => { console.error('Execute error', err); });
+              },
+            );
+            // return (<DialogTitle> successfully added to Google Calendar</DialogTitle>)
           },
-          recurrence: [
-            'RRULE:FREQ=WEEKLY;COUNT=10',
-          ],
-        };
-        events.push(event);
-      },
-    );
-    return events;
+          (err) => { console.error('Error signing in', err); });
+      });
+    });
+  };
+
+  const loadGapiAndAfterwardsInitAuth = () => {
+    const script = document.createElement('script');
```

Comment:
I have absolutely no idea if this is something that we should be doing or not. @Joonpark13 does this look familiar to what you did before?

Reply:
Yeah this is fine